webpack.config.js

```diff
@@ -0,0 +1,223 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+// generated on 2017-09-01 using generator-openmrs-owa 0.6.0
+
+const webpack = require('webpack');
+const path = require('path');
+const fs = require('fs');
+const env = require('yargs').argv.mode;
+const target = require('yargs').argv.target;
+const targetPort = require('yargs').argv.targetPort;
+
+const UglifyPlugin = webpack.optimize.UglifyJsPlugin;
+const CommonsChunkPlugin = webpack.optimize.CommonsChunkPlugin;
+const DedupePlugin = webpack.optimize.DedupePlugin;
+
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const BrowserSyncPlugin = require('browser-sync-webpack-plugin');
+const CopyWebpackPlugin = require('copy-webpack-plugin');
+const WebpackOnBuildPlugin = require('on-build-webpack');
+
+const nodeModulesDir = path.resolve(__dirname, '../node_modules');
+
+require.extensions['.webapp'] = function(module, filename) {
+  module.exports = fs.readFileSync(filename, 'utf8');
+};
+const manifest = require('./app/manifest.webapp');
+
+const THIS_APP_ID = 'openmrs-owa-platformui';
+
+const THIS_APP_VERSION = JSON.parse(manifest).version;
+
+let plugins = [];
+const nodeModules = {};
+
+let outputFile = `.bundle`;
+let vendorOutputFile;
+let outputPath;
+
+let devtool;
+
+let getConfig = function() {
+  let config;
+
+  try {
+    // look for config file
+    config = require('./config.json');
+  } catch (err) {
+    // create file with defaults if not found
+    config = {
+      LOCAL_OWA_FOLDER: '/Users/dondrzzy/openmrs/server1/owa/',
```

Comment:
Do consider it right to use this path, `/Users/dondrzzy/openmrs/server1/owa/`, specific to your developement environment  here. I think it should be relative to the root directory of the project folder. Look into this.

Reply:
Thank you @emmabaye, but this LOCAL_OWA_FOLDER path changes depending on where you are going to deploy this app using the reference application of OpenMRS or the platform or core. It also depends on the machine, whether its a Linux machine or windows machine so someone must follow the `README` guide to set up the webpack.config file properly. And that is the project directory for an Instance of the platform set up using the OpenMRS SDK on a mac/Linux machine. I came to terms with it after looking at how most OWAs on OpenMRS are doing it. I hope this adds some context.